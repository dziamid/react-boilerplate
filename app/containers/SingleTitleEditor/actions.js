/*
 *
 * SingleTitle actions
 *
 */

import {
  UPDATE_TITLE,
} from './constants';

export function defaultAction() {
  return {
    type: UPDATE_TITLE,
  };
}
