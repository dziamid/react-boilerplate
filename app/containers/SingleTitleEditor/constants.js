/*
 *
 * SingleTitle constants
 *
 */

export const DEFAULT_ACTION = 'app/SingleTitle/DEFAULT_ACTION';
