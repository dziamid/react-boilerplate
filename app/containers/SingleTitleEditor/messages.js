/*
 * SingleTitle Messages
 *
 * This contains all the text for the TitlesEditor component.
 */
import { defineMessages } from 'react-intl';
import startCase from 'lodash/startCase';

const messages = {};

['seniority', 'editRelations', 'jobTitle', 'proximity', 'delete'].forEach(t => {
  messages[t] = {
    id: `app.containers.SingleTitle.${t}`,
    defaultMessage: startCase(t),
  };
});

export default defineMessages(messages);
