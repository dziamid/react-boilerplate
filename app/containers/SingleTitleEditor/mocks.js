export const seniorities = [
  { name: 'A', value: 1 },
  { name: 'B', value: 2 },
  { name: 'C', value: 3 },
  { name: 'D', value: 4 },
  { name: 'E', value: 5 },
];

export const proximities = [
  { name: 'AA', value: 1 },
  { name: 'BB', value: 2 },
  { name: 'CC', value: 3 },
  { name: 'DD', value: 4 },
  { name: 'EE', value: 5 },
];

export const titleName = 'Web Developer';
