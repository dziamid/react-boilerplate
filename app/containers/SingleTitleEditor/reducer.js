/*
 *
 * SingleTitle reducer
 *
 */

import { fromJS } from 'immutable';
import {
  UPDATE_TITLE,
} from './constants';

const initialState = fromJS({});

function SingleTitleReducer(state = initialState, action) {
  switch (action.type) {
    case UPDATE_TITLE:
      return state.set();
    default:
      return state;
  }
}

export default SingleTitleReducer;
