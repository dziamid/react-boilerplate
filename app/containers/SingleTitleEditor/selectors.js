import { createSelector } from 'reselect';

/**
 * Direct selector to the SingleTitle state domain
 */
const selectSingleTitleDomain = () => (state) => state.get('SingleTitle');

/**
 * Other specific selectors
 */


/**
 * Default selector used by SingleTitle
 */

const selectSingleTitle = () => createSelector(
  selectSingleTitleDomain(),
  (substate) => substate.toJS()
);

export default selectSingleTitle;
export {
  selectSingleTitleDomain,
};
