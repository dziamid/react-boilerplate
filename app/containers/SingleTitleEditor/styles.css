.SingleTitle { /* stylelint-disable */

}

.formRow {
  display: flex;
  justify-content: space-between;
}

.paper {
  padding: 20px;
  min-width: 90%;
  display: inline-block;
}

.tableWrapper {
  /*margin-top: 50px;*/
  padding-top: 20px;
  max-height: 300px;
  overflow: scroll;
}

.tableRow {
  display: flex;
  justify-content: space-between;
  flex-direction: row;
  border-bottom: 1px dashed gray;
}

.tableHeader {
  composes: tableRow;
  font-weight: 900;
  border-bottom: none;
}

.column {
  width: 25%;
  margin: 10px;
}

.jobTitleCol {
  composes: column;
  width: 45%;
}

.proximityCol{
  composes: column;
  width: 40%;
}

.proximityField {
  max-width: 120px;

}

.buttonCol{
  composes: column;
  width: 10%;
}