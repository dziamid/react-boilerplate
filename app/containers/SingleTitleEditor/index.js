/*
 *
 * SingleTitle
 *
 */

import React, { Component, PropTypes } from 'react';
import { FormattedMessage } from 'react-intl';
import messages from './messages';
import styles from './styles.css';
import MenuItem from 'components/MenuItem';
import H3 from 'components/H3';
import seniorities from 'mocks/seniorities';
import proximities from 'mocks/proximities';
import MUISelectField from 'material-ui/SelectField';
import { difference, compact } from 'lodash';
import RemoveIcon from 'material-ui/svg-icons/content/remove-circle-outline';
import Button from 'components/Button';
import { List } from 'immutable';

import {
  Table, TableBody, TableHeader, TableHeaderColumn, TableRow, TableRowColumn,
}
  from 'material-ui/Table';

export default class SingleTitleEditor extends Component {

  static propTypes = {
    title: PropTypes.object,
    relations: PropTypes.oneOfType([PropTypes.instanceOf(List), PropTypes.array]),
    onSeniorityChange: PropTypes.func,
    onProximityChange: PropTypes.func,
    onRelationRemove: PropTypes.func,
  };

  constructor(props) {
    super(props);

    this.handleSeniorityChange = this.handleSeniorityChange.bind(this);
    this.handleProximityChange = this.handleProximityChange.bind(this);
    this.handleRelationRemove = this.handleRelationRemove.bind(this);
  }

  handleProximityChange(relation, value) {
    this.props.onProximityChange(relation, value);
  }

  handleSeniorityChange(e, key, value) {
    this.props.title.seniority = value;

    this.props.onSeniorityChange(this.props.title.id, value);
  }

  handleRelationRemove(relation) {
    this.props.onRelationRemove(relation);
  }

  render() {
    const {
      title,
      relations,
    } = this.props;

    const seniorityOptions = seniorities.map(s => <MenuItem key={s.value} value={s.value} primaryText={s.name} />);
    const proximityOptions = proximities.map(p => <MenuItem key={p.value} value={p.value} primaryText={p.name} />);

    return (
      <div className={styles.SingleTitle}>
        <H3>
          <div>{title.name}</div>
        </H3>
        <div className={styles.formRow}>
          <MUISelectField
            label={<FormattedMessage {...messages.seniority} />}
            value={title.seniority}
            className={styles.filterField}
            fullWidth
            onChange={this.handleSeniorityChange}
          >

            {seniorityOptions}
          </MUISelectField>
        </div>

        <div className={styles.formRow}>
          <div>
            <H3>
              {<FormattedMessage {...messages.editRelations} />}
            </H3>
          </div>
        </div>
        <Table selectable={false}>
          <TableHeader displaySelectAll={false} adjustForCheckbox={false}>
            <TableRow>
              <TableHeaderColumn className={styles.titleColumn}>
                <FormattedMessage {...messages.jobTitle} />
              </TableHeaderColumn>
              <TableHeaderColumn className={styles.proximityColumn}>
                <FormattedMessage {...messages.proximity} />
              </TableHeaderColumn>
              <TableHeaderColumn />
            </TableRow>
          </TableHeader>
          <TableBody displayRowCheckbox={false}>
            {relations.map((rel, index) =>
              <TableRow key={index}>
                <TableRowColumn className={styles.titleColumn}>
                  {rel.name}
                </TableRowColumn>
                <TableRowColumn className={styles.proximityColumn}>
                  <MUISelectField
                    value={rel.proximity}
                    className={styles.proximityField}
                    onChange={(e, key, value) => this.handleProximityChange(rel, value)}
                  >
                    {proximityOptions}
                  </MUISelectField>

                </TableRowColumn>
                <TableRowColumn className={styles.removeColumn}>
                  <Button
                    icon={<RemoveIcon />}
                    onClick={() => this.handleRelationRemove(rel)}
                  />
                </TableRowColumn>
              </TableRow>
            )}

          </TableBody>
        </Table>

      </div>
    );
  }
}
