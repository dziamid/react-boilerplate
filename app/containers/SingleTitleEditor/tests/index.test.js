import React from 'react';
import SingleTitleEditor from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

describe('<SingleTitleEditor />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <SingleTitleEditor />
    );

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

