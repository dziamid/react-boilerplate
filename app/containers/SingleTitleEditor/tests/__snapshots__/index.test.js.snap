exports[`<SingleTitleEditor /> should render with default props 1`] = `
<div>
  <H3>
    <div />
  </H3>
  <div>
    <SelectField
      autoWidth={false}
      disabled={false}
      fullWidth={true}
      label={
        <FormattedMessage
          defaultMessage="Seniority"
          id="app.containers.SingleTitle.seniority"
          tagName="span"
          values={Object {}} />
      }
      onChange={[Function]}>
      <MenuItem
        primaryText="Entry/Junior Level"
        value={1} />
      <MenuItem
        primaryText="Experienced/Intermediate Level"
        value={2} />
      <MenuItem
        primaryText="Managers and Senior Technical Professionals"
        value={3} />
      <MenuItem
        primaryText="Middle Management"
        value={4} />
      <MenuItem
        primaryText="Senior Management"
        value={5} />
    </SelectField>
  </div>
  <div>
    <div>
      <H3>
        <FormattedMessage
          defaultMessage="Edit Relations"
          id="app.containers.SingleTitle.editRelations"
          tagName="span"
          values={Object {}} />
      </H3>
    </div>
  </div>
  <Connect(ReduxForm)
    fields={
      Array [
        "title",
      ]
    }
    form="SingleTitleEditor" />
  <Table
    allRowsSelected={false}
    fixedFooter={true}
    fixedHeader={true}
    height="inherit"
    multiSelectable={false}
    selectable={false}>
    <TableHeader
      adjustForCheckbox={false}
      displaySelectAll={false}
      enableSelectAll={true}
      selectAllSelected={false}>
      <TableRow
        displayBorder={true}
        hoverable={false}
        hovered={false}
        selectable={true}
        selected={false}
        striped={false}>
        <TableHeaderColumn>
          <FormattedMessage
            defaultMessage="Job Title"
            id="app.containers.SingleTitle.jobTitle"
            tagName="span"
            values={Object {}} />
        </TableHeaderColumn>
        <TableHeaderColumn>
          <FormattedMessage
            defaultMessage="Proximity"
            id="app.containers.SingleTitle.proximity"
            tagName="span"
            values={Object {}} />
        </TableHeaderColumn>
      </TableRow>
    </TableHeader>
    <TableBody
      allRowsSelected={false}
      deselectOnClickaway={true}
      displayRowCheckbox={false}
      multiSelectable={false}
      preScanRows={true}
      selectable={true}
      style={Object {}} />
  </Table>
</div>
`;
